test/validator/gitignore_test.dart

```diff
@@ -143,4 +143,34 @@ void main() {
       workingDirectory: packageRoot,
     );
   });
+
+  test(
+      'Should consider symlinks to be valid files and not list '
+      'them as gitignored', () async {
+    final git = d.git(appPath, [
+      ...d.validPackage().contents,
+      d.dir('dir_with_symlink', [
+        d.file('.pubignore', 'symlink'),
+      ]),
+    ]);
+    await git.create();
+    final packageRoot = p.join(d.sandbox, appPath);
+    await pubGet(
+      workingDirectory: packageRoot,
+    );
+    await d
+        .link(
+          p.join(d.sandbox, appPath, 'dir_with_symlink', 'symlink'),
```

Comment:
```suggestion
          p.join(appPath, 'dir_with_symlink', 'symlink'),
```

Is that not sufficient?

Reply:
Nope, `appPath` is just the directory name of the app directory